src/Listeners/UpdateSplitTitleAfterDiscussionWasRenamed.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/*
+* This file is part of flagrow/flarum-ext-split.
+*
+* Copyright (c) Flagrow.
+*
+* http://flagrow.github.io
+*
+* For the full copyright and license information, please view the license.md
+* file that was distributed with this source code.
+*/
+
+namespace Flagrow\Split\Listeners;
+
+use Illuminate\Contracts\Events\Dispatcher;
+use Flarum\Event\DiscussionWasRenamed;
+
+use Flarum\Core\Repository\PostRepository;
+
+class UpdateSplitTitleAfterDiscussionWasRenamed {
+    /**
+     * @param Dispatcher $events
+     */
+    public function subscribe(Dispatcher $events)
+    {
+        $events->listen(DiscussionWasRenamed::class, [$this, 'whenDiscussionWasRenamed']);
+    }
+
+    /**
+     * @param \Flarum\Event\DiscussionWasRenamed $event
+     */
+    public function whenDiscussionWasRenamed(DiscussionWasRenamed $event) {
+            
+        $this->posts = app(PostRepository::class);
+        $allPosts = $this->posts->
+                query()->
+                where('type',"=","discussionSplit")->
+                where('content',"like","%".$event->discussion->id."-%")->
+                take(10)->
```

Comment:
This limits the number 10. Now we can assume there won't be more of them. But I still dislike any hardcoded limitation. In order to circumvent this use the `chunk()` method.

See: https://laravel.com/docs/5.1/queries#retrieving-results (chunking)

Reply:
indeed! I did not know multiple splits can be linked to one discussions.